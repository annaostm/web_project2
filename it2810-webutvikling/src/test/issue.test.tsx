import { unmountComponentAtNode } from 'react-dom';
import renderer from 'react-test-renderer';
import Issue from '../components/issue/issue'

let container: Element | DocumentFragment | null = null;

beforeEach(() => {
  // setup a DOM element as a render target
  container = document.createElement("div");
  document.body.appendChild(container);
});

afterEach(() => {
  // cleanup on exiting
  if(container == null){
    throw "container is null"
  }
  unmountComponentAtNode(container);
  container = null;
});
const b = renderer.create(<Issue data={[{ merged: true, branchDate: "200300", id: "1234", title: "issue", state: "done", created_at: "200300" }]} fromDateFilter={"28122017"} toDateFilter={"22122018"} authorFilter={"Hedda Friis"}></Issue>);

it('renders correctly', () => {
    const tree= b.toJSON();
    expect(tree).toMatchSnapshot();
  });