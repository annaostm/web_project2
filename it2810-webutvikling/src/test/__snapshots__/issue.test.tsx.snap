// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <h1>
    Issues
  </h1>
  <p>
    No issues to show.
  </p>
</div>
`;
